pkg/ipam/service/ipallocator/allocator.go

```diff
@@ -68,13 +68,21 @@ type Range struct {
 // the backing store.
 func NewCIDRRange(cidr *net.IPNet) *Range {
 	base := bigForIP(cidr.IP)
-	max := maximum(0, int(RangeSize(cidr)-2)) // don't use the network broadcast,
+	size := RangeSize(cidr)
+
+	// for any CIDR other than /32 or /128:
+	if size > 1 {
```

Comment:
Can we add a unit test for this? And maybe also add a comment in the godoc?

Reply:
We could also allow this for /31 or /127 ranges too btw - those are also empty otherwise